src/backend/src/app.config.ts

```diff
@@ -0,0 +1,46 @@
+import { NestFactory } from '@nestjs/core';
+import {
+  ExpressAdapter,
+  NestExpressApplication,
+} from '@nestjs/platform-express';
+import * as express from 'express';
+import * as morgan from 'morgan';
+import { AppModule } from './app.module';
+import { SwaggerDocs } from './common/swagger';
+
+export async function createNestApp(): Promise<{
+  app: NestExpressApplication;
+  expressApp: express.Application;
+}> {
+  // Express App
+  const expressApp = express();
+  expressApp.disable('x-powered-by');
+
+  // Nest Application with Express Adapter
+  const app: NestExpressApplication =
+    await NestFactory.create<NestExpressApplication>(
+      AppModule,
+      new ExpressAdapter(expressApp),
+      {
+        logger:
+          process.env.NODE_ENV === 'development'
+            ? ['error', 'warn', 'log', 'verbose', 'debug']
+            : ['log', 'warn', 'error'],
+      },
+    );
+
+  // append request logs
+  app.use(morgan('tiny'));
+
+  app.setGlobalPrefix('/api');
+
+  // Enable Swagger for non prod environment
```

Comment:
If we're developing a public API, I'd love for our swagger to be readily available in PROD.

What do you think?

Reply:
I mean, sure if we need it to be public. Are you talking from a sense that this repo is public and so people should have access to the API documentation too? If so, sure - I will go ahead and make that change 👍 You have better context than me.

However, in that case if we are having our APIs and documentation public - we should have API throttling set up - where someone maliciously do not try to bring our system down by continuously hitting the endpoints / bots. Thoughts?